app/database/operator/server_data_operator/handlers/thread_in_team.ts

```diff
@@ -40,11 +40,11 @@ const ThreadInTeamHandler = (superclass: any) => class extends superclass {
                 const newValue = {
                     thread_id: thread.id,
                     team_id: teamId,
-                    loaded_in_global_threads: thread.loaded_in_global_threads,
+                    loaded_in_global_threads: Boolean(loadedInGlobalThreads),
                 };
 
-                // update record if loaded_in_global_threads is true
-                if (chunk && thread.loaded_in_global_threads) {
+                // update record only if loaded_in_global_threads is true
+                if (chunk && loadedInGlobalThreads) {
```

Comment:
If I understood it correctly, at this point, a user is following a thread and now, we are showing that thread under the 'Global Thread'?

Reply:
Hm, it's a bit more complex.
So in global threads we have two lists, unread threads, and all threads.
`loadedInGlobalThreads` represents all threads.

When we load unread threads we can't add them in the all threads list since it might create gaps in pagination,
so we add threads in all threads when we explicitly ask for those. 
All others threads will have this set as false.
Now ,the threads_in_team table is an intermediary, holding the relation of team and thread id for those 2 lists.
both those columns are immutable, they never change, you can either create or delete.
However, we also have the `loaded_in_global_threads` column, which can be set from `false` to `true`,
but not the other way around.